### ISSUE TITLE
extract_corpora: Invalid book ID causing multiple books not to be extracted from the project

### ISSUE THREAD

**Issue by mmartin9684-sil** [sillsdev/silnlp#574]
A project (Galego_2024_10_24.save) has an invalid book ID in the "\id" tag ("\id JUD") for the book of Judges; it should be "JDG" instead of "JUD". 

No error was reported for this mismatch, but the book content was not extracted into the extract file.  Also, most other books in the project were not extracted into the extract file either.  The extract file contained verse extracts only for GEN, JUD, and REV, even though the project has a complete NT and multiple OT books.

A warning about the book ID mistake would be helpful, and the error should not affect the extraction of the other books.

**Comment by ddaspit** [comment]
@mshannon-sil I agree with your analysis. When there is a mismatch between the file name and the "\id" tag, we don't know which one is correct. We could decide to always use the book code from the file name and ignore the "\id" tag and that would fix the issue in this case, but in other cases, it might be the opposite. The safest thing to do is to throw an exception.

**Comment by isaac091** [comment]
@mshannon-sil can we close this issue?

**Comment by mshannon-sil** [comment]
Not yet, it looks like the last release of the python package for machine.py was in October, and the fix was issued in November. @johnml1135 is there a plan for a new release of machine.py relatively soon?

**Comment by johnml1135** [comment]
Just updated to 1.4.0.

**Comment by mshannon-sil** [comment]
@johnml1135 on pypi, the latest version still seems to be 1.3.0 for sil-machine: https://pypi.org/project/sil-machine/#history

**Comment by johnml1135** [comment]
Ugh.  The build crashed.  It should be resolved today or tomorrow.
